snap/validate.go

```diff
@@ -153,10 +162,15 @@ func validateSocketAddrNetHost(socket *SocketInfo, fieldName string, address str
 }
 
 func validateSocketAddrNetPort(socket *SocketInfo, fieldName string, port string) error {
-	if _, err := strconv.ParseUint(port, 10, 16); err != nil {
-		return fmt.Errorf("socket %q has invalid %q port number %q", socket.Name, fieldName, port)
+	var val uint64
+	var err error
+	retErr := fmt.Errorf("socket %q has invalid %q port number %q", socket.Name, fieldName, port)
```

Comment:
`invalid %q port number %q` -- what will be the value of the first %q (field name)?

Reply:
it's the name of the field under validation, specifically "listen-stream"